ocaml/xapi/xapi_pvs_cache.ml

```diff
@@ -21,19 +21,22 @@ exception No_cache_sr_available
 exception No_cache_vdi_present
 
 module VDI = struct
-	let find_all ~__context ~sr =
+	let find_all ~__context ~sr ~site =
 		let open Db_filter_types in
-		Db.VDI.get_refs_where ~__context ~expr:(And
+		match Db.PVS_cache_storage.get_refs_where ~__context ~expr:(And
 			(Eq (Field "SR", Literal (Ref.string_of sr)),
-			Eq (Field "type", Literal "pvs_cache")))
+			Eq (Field "site", Literal (Ref.string_of site))))
+		with
+		| [] -> None, []
+		| cacheSR :: _ -> Some cacheSR, (Db.PVS_cache_storage.get_cache_VDIs ~__context ~self:cacheSR)
 
-	let find_one ~__context ~sr =
-		match find_all ~__context ~sr with
-		| [] -> None
-		| vdi :: _ -> Some vdi
+	let find_one ~__context ~sr ~site ~host =
+		match find_all ~__context ~sr ~site with
+		| (None, _) -> None
+		| (_, vdi_map) -> try Some (List.assoc host vdi_map) with _ -> None
```

Comment:
I think this should match only `Not_found` to avoid that we are masking errors here.